runtime/lua/vim/lsp/protocol.lua

```diff
@@ -0,0 +1,437 @@
+-- Protocol for the Microsoft Language Server Protocol (mslsp)
+
+local util = require('vim.lsp.util')
+local server_config = require('vim.lsp.server_config')
+local protocol = {}
+
+
+protocol.DiagnosticSeverity = {
+  [1] = 'Error',
+  [2] = 'Warning',
+  [3] = 'Information',
+  [4] = 'Hint',
+}
+
+protocol.MessageType = {
+  [1] = 'Error',
+  [2] = 'Warning',
+  [3] = 'Info',
+  [4] = 'Log',
+}
+
+protocol.FileChangeType = {
+  [1] = 'Created',
+  [2] = 'Changed',
+  [3] = 'Deleted',
+}
+
+protocol.CompletionItemKind = {
+  [1] = 'Text',
+  [2] = 'Method',
+  [3] = 'Function',
+  [4] = 'Constructor',
+  [5] = 'Field',
+  [6] = 'Variable',
+  [7] = 'Class',
+  [8] = 'Interface',
+  [9] = 'Module',
+  [10] = 'Property',
+  [11] = 'Unit',
+  [12] = 'Value',
+  [13] = 'Enum',
+  [14] = 'Keyword',
+  [15] = 'Snippet',
+  [16] = 'Color',
+  [17] = 'File',
+  [18] = 'Reference',
+  [19] = 'Folder',
+  [20] = 'EnumMember',
+  [21] = 'Constant',
+  [22] = 'Struct',
+  [23] = 'Event',
+  [24] = 'Operator',
+  [25] = 'TypeParameter',
+}
+
+protocol.CompletionTriggerKind = {
+  [1] = 'Invoked',
+  [2] = 'TriggerCharacter',
+}
+
+protocol.DocumentHighlightKind = {
+  [1] = 'Text',
+  [2] = 'Read',
+  [3] = 'Write',
+}
+
+protocol.SymbolKind = {
+  [1] = 'File',
+  [2] = 'Module',
+  [3] = 'Namespace',
+  [4] = 'Package',
+  [5] = 'Class',
+  [6] = 'Method',
+  [7] = 'Property',
+  [8] = 'Field',
+  [9] = 'Constructor',
+  [10] = 'Enum',
+  [11] = 'Interface',
+  [12] = 'Function',
+  [13] = 'Variable',
+  [14] = 'Constant',
+  [15] = 'String',
+  [16] = 'Number',
+  [17] = 'Boolean',
+  [18] = 'Array',
+}
+
+protocol.errorCodes = {
+  -- Defined by JSON RPC
+  [-32700] = 'Parse error',
+  [-32600] = 'Invalid Request',
+  [-32601] = 'Method not found',
+  [-32602] = 'Invalid params',
+  [-32603] = 'Internal error',
+  [-32099] = 'Server Error Start',
+  [-32000] = 'Server Error End',
+  [-32002] = 'Server Not Initialized',
+  [-32001] = 'Unknown Error Code',
+  -- Defined by the protocol
+  [-32800] = 'Request Cancelled',
+}
+
+
+protocol.TextDocumentSaveReason = {
+  [1] = 'Manual',
+  [2] = 'AfterDelay',
+  [3] = 'FocusOut',
+}
+
+-- Helper functions
+local check_table = function (t)
+  if type(t) ~= 'table' then
```

Comment:
this could implicitly hide type/argument errors. Probably we should just allow converting `nil` (i e no arg) to empty dict, but nothing else. I e check `t == nil` instead.

Reply:
And in general, throw `error()` if the type is wrong / not supported.